packages/react-dom/src/client/ReactDOMHostConfig.js

```diff
@@ -374,6 +379,25 @@ export function getCurrentEventPriority(): * {
   return getEventPriority(currentEvent.type);
 }
 
+export function logRecoverableError(
+  config: ErrorLoggingConfig,
+  error: mixed,
+): void {
+  const onRecoverableError = config;
+  if (onRecoverableError !== null) {
```

Comment:
This feels like a reconciler specific concern to me. Especially since this is likely to grow with other types of callbacks that will have more advanced semantics than this callback will handle.

One case I'd expect we might want is to scope this to subtree. Like error boundaries but for recoverable errors.

The reconciler can make this choice and schedule an IdlePriority callback itself if it wants to. Although it feels a bit presumptuous to do that for the user since you can schedule it in the callback. Although paternalism is on brand for us I guess.

Reply:
The main reasons

Reply:
Yeah I went back and forth on that. The main reason I ended up adding a host config is because it needs a default fallback behavior; didn't feel right to assume that throwing in a Scheduler task is necessarily the correct default for each environment (although I suppose that's what we do for uncaught errors). An alternative is to resolve the default option within the renderer before passing it to the root constructor but that feels like a superficial difference. Don't mind either way.

Related comment: https://github.com/facebook/react/blob/e6a1df0ac2fa896aa516684539760a186f115325/packages/react-reconciler/src/ReactFiberRoot.new.js#L114-L117

Reply:
Yea I don't think it's always correct to throw in a task so having host config makes sense for the default fallback behavior. It's the main API behavior that doesn't seem right to determine in the host config. In other words, resolving the default option within the renderer would make sense. That also avoids leaking the type to all the host configs.